File: Java/Class-2.4/src/org/launchcode/main/Temperature.java
package org.launchcode.main;

// 5.1.2 Static Fields
public class Temperature {
	private double fahrenheit;

	public static double absoluteZeroFahrenheit = -459.67;

	public double getFahrenheit() {
		return fahrenheit;
	}

	public void setFahrenheit(double fahrenheit) {

		if (fahrenheit < absoluteZeroFahrenheit) {
			throw new IllegalArgumentException("Value is below absolute zero");
		}

		this.fahrenheit = fahrenheit;
	}

	public void printAbsoluteZero() {
		// Both of these are wrong!
		System.out.println("Absolute zero in F is: " + absoluteZeroFahrenheit);

		// We can also be more explicit - This is wrong, don't do this...
		System.out.println("Absolute zero in F is: " + this.absoluteZeroFahrenheit);
	}
}

File: CSharp/LC101-Unit2/Class-2.5/Temperature.cs
using System;
namespace Class_2._5
{
    public class Temperature
    {
        private double fahrenheit;
        public static double AbsoluteZeroFahrenheit { get; } = -459.67;

        public double Fahrenheit
        {
            get
            {
                return fahrenheit;
            }
            set
            {

                if (value < AbsoluteZeroFahrenheit)
                {
                    throw new ArgumentOutOfRangeException("Value is below absolute zero");
                }

                fahrenheit = value;
            }
        }
    }
}
